### PR TITLE
[bndtools] launch m2e

### PR DESCRIPTION
**Pull request by rotty3000**

### CODE REVIEW

**Comment by rotty3000** [comment]
@bjhargrave I left the full index in there just so you could play with the PR if you needed. Did we really need EMF btw?

FYI, the first commit is pure sort. I did this so that I could use diff on the results as I iterated to get this working. It's much easier to do this with sorted values. Can we keep it sorted?

**Comment by rotty3000** [comment]
FYI, if you try loading an existing workspace from a newer version of eclipse, it won't show any projects. You need to re-import the projects and then everything seemed to work fine.

File: bndtools.core/bndtools.gtk.linux.x86_64.bndrun [hunk 1]
```diff
@@ -1,15 +1,24 @@
 -include: bndtools.shared.bndrun
 
 -runbundles.linux: \
+	org.eclipse.swt.gtk.linux.x86_64,\
+	org.eclipse.e4.ui.workbench.renderers.swt,\
```
Comment (bjhargrave) [inline bndtools.core/bndtools.gtk.linux.x86_64.bndrun]: this is already in the shared bndrun, so don't duplicate it here.

File: bndtools.core/bndtools.gtk.linux.x86_64.bndrun [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+	${crossplatform.vmargs}
 
 -runproperties: \
 	${crossplatform.runprops},\
 	osgi.ws=gtk,\
 	osgi.arch=x86_64,\
 	osgi.os=linux
+
+-runblacklist.linux: \
+	osgi.identity;filter:='(osgi.identity=*macosx*)'
```
Comment (bjhargrave) [inline bndtools.core/bndtools.gtk.linux.x86_64.bndrun]: Shouldn't `*win32*` also be in here?

File: bndtools.core/bndtools.shared.bndrun [hunk 3]
```diff
@@ -13,46 +13,9 @@ crossplatform.runprops: \
...
-	bnd.identity;id='org.eclipse.emf.codegen.ecore.ui',\
-	bnd.identity;id='org.eclipse.emf.codegen.ui',\
-	bnd.identity;id='org.eclipse.emf.common.ui',\
-	bnd.identity;id='org.eclipse.equinox.event',\
-	bnd.identity;id='org.eclipse.ant.core',\
-	bnd.identity;id='org.eclipse.ant.ui',\
-	bnd.identity;id='org.eclipse.ui.ide.application',\
-	bnd.identity;id='org.eclipse.ui.ide',\
-	bnd.identity;id='org.eclipse.sdk',\
-	bnd.identity;id='org.eclipse.pde.ua.ui',\
```
Comment (bjhargrave) [inline bndtools.core/bndtools.shared.bndrun]: why did you pull pde? I think it should remain for the bndtools.pde project, no?

File: cnf/ext/repositories.bnd [hunk 4]
```diff
@@ -37,6 +37,11 @@ mavencentral:           https://repo.maven.apache.org/maven2
         locations="https://dl.bintray.com/bndtools/eclipse-repo/4.10/index.xml.gz";\
         poll.time=-1;\
         cache="${workspace}/cnf/cache/stable/Eclipse-2018-12",\
+    aQute.bnd.repository.osgi.OSGiRepository;\
```
Comment (bjhargrave) [inline cnf/ext/repositories.bnd]: Obviously we don't want to commit this. Are there any artifacts missing from the bintray repo I need to add?

File: bndtools.core/bndtools.gtk.linux.x86_64.bndrun (hunk 1 above)
Comment (rotty3000) [reply bndtools.core/bndtools.gtk.linux.x86_64.bndrun]: right.

File: bndtools.core/bndtools.gtk.linux.x86_64.bndrun (hunk 2 above)
Comment (rotty3000) [reply bndtools.core/bndtools.gtk.linux.x86_64.bndrun]: added for completeness though it was never getting resolved.

File: bndtools.core/bndtools.shared.bndrun (hunk 3 above)
Comment (rotty3000) [reply bndtools.core/bndtools.shared.bndrun]: re-added!

File: cnf/ext/repositories.bnd (hunk 4 above)
Comment (rotty3000) [reply cnf/ext/repositories.bnd]: Yes, there are arfiacts missing from the slim repo, see the diff. I removed the repo reference.

**Pull request merged**